spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubMessageSource.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.integration.inbound;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.cloud.gcp.pubsub.core.subscriber.PubSubSubscriberOperations;
+import org.springframework.cloud.gcp.pubsub.integration.AckMode;
+import org.springframework.cloud.gcp.pubsub.integration.PubSubHeaderMapper;
+import org.springframework.cloud.gcp.pubsub.support.GcpPubSubHeaders;
+import org.springframework.cloud.gcp.pubsub.support.converter.ConvertedAcknowledgeablePubsubMessage;
+import org.springframework.integration.endpoint.AbstractMessageSource;
+import org.springframework.integration.mapping.HeaderMapper;
+import org.springframework.util.Assert;
+
+/**
+ * A <a href="https://cloud.google.com/pubsub/docs/pull#pubsub-pull-messages-sync-java">PubSub Synchronous pull</a>
+ * implementation of {@link AbstractMessageSource}.
+ *
+ * Enables more even load balancing with polling adapters, at the cost of efficiency gained from batched pulls.
+ *
+ * @author Elena Felder
+ *
+ * @since 1.1
+ */
+public class PubSubMessageSource extends AbstractMessageSource<Object> {
```

Comment:
Just wanted to clarify on naming convention - does MessageSource imply "polling" and channelAdapter imply "stream" in Spring? If not I would suggest including 'sync' or 'polling' in the class name.

Reply:
That's correct. The `MessageSource` does nothing by itself. Only wrapping `SourcePollingChannelAdapter` does the polling magic.
I agree that `PubSubInboundChannelAdapter` is not so obvious name, but our convention to use such a name for those components which are doing the streaming via listening in the internal task.
Sometimes we call them `MessageDrivenEndpoint`, e.g. `JmsMessageDrivenEndpoint`.